qemu/tests/virtio_fs_daemon_readonly.py

```diff
@@ -0,0 +1,146 @@
+        fs = params["filesystems"]
+        fs_params = params.object_params(fs)
+
+        fs_target = fs_params["fs_target"]
+        fs_dest = fs_params["fs_dest"]
+
+        if os_type == "linux":
+            utils_misc.make_dirs(fs_dest, session)
+            error_context.context(
+                "Mount virtiofs target %s to %s inside"
+                " guest." % (fs_target, fs_dest),
+                test.log.info,
+            )
+            if not utils_disk.mount(fs_target, fs_dest, "virtiofs", session=session):
+                utils_misc.safe_rmdir(fs_dest, session=session)
+                test.fail("Failed to mount virtiofs {fs_target}.")
+        else:
+            error_context.context("Start virtiofs service in guest.", test.log.info)
+            debug_log_operation = params.get("debug_log_operation")
+            if debug_log_operation:
+                session = virtio_fs_utils.operate_debug_log(
+                    test, params, session, vm, debug_log_operation
+                )
+            virtio_fs_utils.start_viofs_service(test, params, session)
+
+            fs_dest = "%s:" % virtio_fs_utils.get_virtiofs_driver_letter(
+                test, fs_target, session
+            )
+
+        guest_mnts[fs_target] = fs_dest
+        return session
+
+    def stop_service(session):
+        error_context.context("Stop virtiofs service in guest.", test.log.info)
+
+        if os_type == "linux":
+            for fs_target, fs_dest in guest_mnts.items():
+                utils_disk.umount(fs_target, fs_dest, "virtiofs", session=session)
+                utils_misc.safe_rmdir(fs_dest, session=session)
+        else:
+            if guest_mnts:
+                virtio_fs_utils.stop_viofs_service(test, params, session)
+        session.close()
+
+    def test_readonly():
+        def check_file_content():
+            error_context.context(
+                "Check file content after mounting.",
+                test.log.info,
+            )
+
+            for fs_dest in guest_mnts.values():
+                out = session.cmd_output(params["read_file_cmd"] % fs_dest).strip()
+                test.log.debug("File content: %s", out)
+                if out != params["test_data"]:
+                    test.fail(f"Wrong file content found: {out}")
+
+        def write_file():
+            error_context.context(
+                "Write the file after mounting.",
+                test.log.info,
+            )
+
+            for fs_dest in guest_mnts.values():
+                output = session.cmd_output(params.get("write_file_cmd") % fs_dest)
+                test.log.info(output)
+                if os_type == "linux":
+                    if params.get("check_str_linux") not in output:
+                        test.fail("The mounted dir is not read-only.")
+                else:
+                    if params.get("check_str_win") not in output:
+                        test.fail("The mounted dir is not read-only.")
```

Comment:
we may define one check_str param in cfg, e.g. 
check_str = 'Read-only file system'
Windows:
check_str = 'A device attached to the system is not functioning'

Then here 'if else' is not needed

Reply:
Sure, thanks for the catch.